theseus/embodied/kinematics/robot_model.py

```diff
@@ -27,12 +31,12 @@ class IdentityModel(RobotModel):
     def __init__(self):
         super().__init__()
 
-    def forward_kinematics(self, robot_pose: LieGroup) -> torch.Tensor:
+    def forward_kinematics(self, robot_pose: RobotModelInput) -> RobotModelOuput:
         if isinstance(robot_pose, SE2):
-            return robot_pose.translation.data.view(-1, 2, 1)
+            return robot_pose.translation
```

Comment:
Should we remove this condition since the input will not be SE2?

Reply:
Fixed.

Reply:
Actually a question for @exhaustin: how do you handle an input space in DRM for a mobile manipulator which would be base + arm (SE2 x Vector)?

Reply:
Regarding `Point2`, we currently allow `Vector` inputs to the robot model.

Reply:
Ah yes, but when orientation is involved we would need SE2. We don't need to have all the variants implemented but thinking about the API would be useful so we don't have to refactor later. Here are some possible types of robot bodies with their input space:
- Mobile base: SE2
- Arm(N): Vector(N) [we are currently focusing on this]
- Arm(N) + hand(H): Vector(N+H)
- Mobile base and arm: SE2 x Vector(N)
- Mobile base, vertical actuator,  arm: SE2 x Vector(N+1)
- Mobile base, vertical actuator,  left arm, right arm : SE2 x Vector(2N+1)
- Aerial robot: SE3
- Aerial robot + arm: SE3 x Vector(N)
- more combinations for legged robots or humanoids or robots with a camera that have a pan/tilt joint.

Reply:
> Actually a question for @exhaustin: how do you handle an input space in DRM for a mobile manipulator which would be base + arm (SE2 x Vector)?

Simulators and kinematics libraries I've worked with seems to be designed for rotary joints (for example, the PyBullet API `setMotorControlArray` makes the assumption that all actuators are rotary motors). I don't have a clear idea of how we could generalize to structured joints that aren't just an array of joint angles, and I can't find any examples out there.

If I imagine how I would set up a problem on mobile manipulators, I would set up the forward kinematics separately as the transform of the mobile base + the transform of the manipulator computed using the urdf. I think this could exist in user code instead of having a single forward kinematics function that encapsulates all special cases you listed above.

Reply:
That makes sense, base transform + FK is the version I am familiar with as well. Let's proceed with the version where URDFRobotModel take a Vector as an input that represents the joint angles of a kinematic chain.

An aside question: does DRM current support branching chains for robot hand fingers? (I don't think this would affect the input since it would still be a Vector)

Reply:
Yes we support kinematic trees as well as chains. The input is still a vector of actuated joints, in the order they are defined in the URDF file.